### DISCUSSION TITLE
Pre-selecting collection based on page field

### DISCUSSION THREAD

**Discussion by ralnss** [wagtail/wagtail#12421] [Q&A]
Hello wagtail-people!

I'm trying to pre-select the collection for images I want to add to my pages. 

My pages have a field linking to a corresponding collection and I already implemented image filtering in the `construct_image_chooser_queryset` hook based on that collection field, but I would like to also preselect that collection for uploading new images in the upload-tab of the image chooser modal. It's also not optimal because I can't select images from other collections any more.

It would be great to have it very much like it is implemented in the admin main menu image page.
Is there a way to achieve that for the image chooser modal on my pages? I'm using both the default image chooser and the `MultipleChooserPanel` for some pages.
In the main images page, a url parameter `?collection_id=X` is added, could this be an approach to try?

There will probably be cases where one page has not only one but multiple linked collections. For this, it would be nice if there was a way to pass a list of collections to the chooser. I'm not sure how that could be solved though.

Any help is appreciated!

**Comment by thibaudcolas** [comment]
hi @ralnss, I’m not aware of a way to do this. That could be a good feature request!

**Reply by ralnss** [reply]
Hi @thibaudcolas, thanks for your reply! Do you think there is a way to filter the list of available collections? Something like a preselection would be great if there is only one relevant associated colection, maybe filtering the list would be a more general approach, maybe even encompassing the preselecting-case.
